samples/openapi3/client/petstore/typescript/builds/default/apis/PetApi.ts

```diff
@@ -364,12 +339,17 @@ export class PetApiRequestFactory extends BaseAPIRequestFactory {
         const requestContext = _config.baseServer.makeRequestContext(localVarPath, HttpMethod.POST);
         requestContext.setHeaderParam("Accept", "application/json, */*;q=0.8")
 
-        // Query Params
-
-        // Header Params
-
         // Form Params
-        let localVarFormParams = new FormData();
+        const useForm = canConsumeForm([
+            'multipart/form-data',
+        ]);
+
+        let localVarFormParams
+        if (useForm) {
+            localVarFormParams = new FormData();
+        } else {
+            localVarFormParams = new URLSearchParams();
+        }
```

Comment:
I'm wondering whether it would make more sense to handle the conditions during code generation? The content types are known from the spec, at generation time. It would make the emitted code simpler.

Reply:
I've thought about it but
We did in a similar way as in [typescript-fetch](https://github.com/OpenAPITools/openapi-generator/pull/3934) and [typescript-angular](https://github.com/OpenAPITools/openapi-generator/blob/da5d611ca98a42f69d21fb844dfb4e26fd558e54/modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache#L84)
I can do it as you proposed 
But I don’t know what would be better 
what do you think? 
CC @amakhrov @macjohnny

Reply:
I believe preferring simplicity in the end code over simplicity in the codegen template is the right tradeoff - hence handling the conditions during code generation is better.
But I see how it can increase the scope of PR here.
I totally don't object keeping it the way it is right now

Reply:
let's leave it the way it is right now
because it will increase the scope of PR here.
CC @amakhrov @macjohnny